go/libraries/doltcore/doltdb/durable/index.go

```diff
@@ -18,20 +18,63 @@ import (
 	"context"
 	"fmt"
 
+	"github.com/dolthub/dolt/go/store/hash"
+
 	"github.com/dolthub/dolt/go/store/types"
 )
 
+type Index interface {
```

Comment:
Presumably this is a stub? Would be nice to understand what the intended final shape is

Reply:
> Presumably this is a stub? Would be nice to understand what the intended final shape is

That would be nice...